### ISSUE TITLE
Update CI systems with new QAT driver

### ISSUE THREAD

**Issue by peluse** [spdk/spdk#886]
There’s a new QAT driver at https://01.org/sites/default/files/downloads//qat1.7.l.4.6.0-00025.tar.gz that actually works.  This is like only the 2nd time I’ve been successful setting one up so I’ll leave it to one of you to update the test systems.  I’ll throw a github issue up for it.  Note that compressdev CI testing (patch pending) is limited to ISAL right now.  Once this new driver is in place we can update compress.sh to teset both ISAL and QAT.  I can do that part if you just let me know once this is done

**Comment by peluse** [comment]
BTW I'm sure you guys have it but here's the directions, note that on my system I no longer needed the sed workourd they mention in another page somewhere: https://doc.dpdk.org/guides/cryptodevs/qat.html#installation-using-01-org-qat-driver

**Comment by karlatec** [comment]
Updated on GP4 for now.
Tests with the same config as in CI passed locally:
```
SPDK_BUILD_DOC=0
SPDK_BUILD_IOAT_KMOD=0
SPDK_BUILD_SHARED_OBJECT=1
SPDK_FEDORA_PACKAGING=0
SPDK_RUN_CHECK_FORMAT=0
SPDK_RUN_SCANBUILD=0
SPDK_RUN_VALGRIND=0
SPDK_RUN_FUNCTIONAL_TEST=1
SPDK_TEST_UNITTEST=0
SPDK_TEST_ISCSI=0
SPDK_TEST_NVME=0
SPDK_TEST_BDEV_OCSSD=0
SPDK_TEST_FTL=0
SPDK_TEST_NVMF=0
SPDK_TEST_RBD=0
SPDK_TEST_VHOST=0
SPDK_TEST_BLOCKDEV=1
SPDK_TEST_IOAT=0
SPDK_TEST_EVENT=0
SPDK_TEST_BLOBFS=0
SPDK_TEST_PMDK=0
SPDK_TEST_LVOL=0
SPDK_TEST_ISAL=1
SPDK_RUN_VPP=0
SPDK_TEST_CRYPTO=1
SPDK_TEST_REDUCE=1
SPDK_TEST_OCF=0
SPDK_RUN_ASAN=0
SPDK_RUN_UBSAN=1
SPDK_TEST_VHOST_INIT=0
```

However I see a lot of error/warning messages when initializing SPDK/DPDP. For example:
```
[This one seems to happen on every init]
Starting 17 threads
Starting SPDK v19.07-pre / DPDK 19.05.0 initialization...
[ DPDK EAL parameters: fio --no-shconf -c 0x1 --log-level=lib.eal:6 --log-level=lib.cryptodev:5 --log-level=user1:6 --base-virtaddr=0x200000000000 --match-allocations --file-prefix=spdk_pid204393 ]
EAL: No available hugepages reported in hugepages-1048576kB
EAL: VFIO support initialized
compdev_isal_create():
ISA-L library version used: 2.24.0
vbdev_compress.c: 330:vbdev_init_compress_drivers: *NOTICE*: created virtual PMD compress_isal
isal_comp_pmd_qp_setup(): Failed to create unique name for isal compression device
vbdev_compress.c: 246:create_compress_dev: *NOTICE*: FYI failed to setup a queue pair on compressdev 48 with error 4294967295 so limiting to 10 qpairs
vbdev_compress.c: 373:vbdev_init_compress_drivers: *NOTICE*: initialized QAT PMD
cryptodev_aesni_mb_create() line 1304: IPSec Multi-buffer library version used: 0.52.0
```

```
[This one looks like from setup.sh or qat_setup.sh]

Restarting all devices.
Processing /etc/c6xx_dev0.conf
Processing /etc/c6xx_dev1.conf
Processing /etc/c6xx_dev2.conf
Checking status of all devices.
There is 3 QAT acceleration device(s) in the system:
 qat_dev0 - type: c6xx,  inst_id: 0,  node_id: 1,  bsf: 0000:84:00.0,  #accel: 5 #engines: 10 state: up
 qat_dev1 - type: c6xx,  inst_id: 1,  node_id: 1,  bsf: 0000:85:00.0,  #accel: 5 #engines: 10 state: up
 qat_dev2 - type: c6xx,  inst_id: 2,  node_id: 1,  bsf: 0000:86:00.0,  #accel: 5 #engines: 10 state: up
failed to start the qat service. Something may be wrong with your 01.org driver.
0000:84:00.0 set to 16 VFs
0000:85:00.0 set to 16 VFs
0000:86:00.0 set to 16 VFs
unbinding driver c6xxvf from qat VF at BDF 0000:84:01.0
unbinding driver c6xxvf from qat VF at BDF 0000:84:01.1
unbinding driver c6xxvf from qat VF at BDF 0000:84:01.2
unbinding driver c6xxvf from qat VF at BDF 0000:84:01.3
unbinding driver c6xxvf from qat VF at BDF 0000:84:01.4
unbinding driver c6xxvf from qat VF at BDF 0000:84:01.5
[...]
unbinding driver c6xxvf from qat VF at BDF 0000:86:02.6
unbinding driver c6xxvf from qat VF at BDF 0000:86:02.7
Properly configured the qat device with driver uio_pci_generic.
```

In the dock pages you linked there is a note:
"In the BIOS ensure that SRIOV is enabled and VT-d is disabled."
I wasn't aware of that. All of our CI systems use VT-d (for vhost tests I guess). I'll try again with that disabled and see if errors persist.

**Comment by karlatec** [comment]
Same thing with VT-d disabled. Let me know if I should proceed with update on other QAT system.

**Comment by peluse** [comment]
Awesome thanks.  The error messages you are seeing are semi-unrelated and I have a DPDK patch on our fork to address them. When this merges https://review.gerrithub.io/c/spdk/spdk/+/463548 they will go away.  I'll ask Fiona about the VTd thing

**Comment by karlatec** [comment]
Im going to run tests via CI instead of just manually to confirm if all is OK, and proceed with the second system.

**Comment by peluse** [comment]
thanks!

**Comment by karlatec** [comment]
Forgot to add a comment, sorry.
Both platforms have the driver updated now. Please confirm all is OK and then we can close the issue:)

**Comment by peluse** [comment]
OK, I need to make one test script update to actually use QAT cards for compress (right now ISAL is hardcoded).  I'll make that change and add yo;u as a reviewer so you see it
